backend/app/dependencies.py

```diff
@@ -10,3 +18,49 @@ async def only_for_admin(authenticated_user: User = Depends(get_authenticated_us
             status_code=status.HTTP_403_FORBIDDEN,
             detail="Only admin can execute this operation",
         )
+
+
+class OnlyProductRoles:
+    """Only allow users that have a membership of a certain role type to this product"""
+
+    def __init__(self, allowed_roles: list[DataProductUserRole]):
+        if allowed_roles == [DataProductUserRole.MEMBER]:
+            allowed_roles.append(DataProductUserRole.OWNER)
+        self.allowed_roles = allowed_roles
+
+    def __call__(
+        self,
+        data_product_name: Optional[str] = None,
+        id: Optional[UUID] = None,
+        authenticated_user: User = Depends(get_authenticated_user),
+        db: Session = Depends(get_db_session),
+    ) -> None:
+        if id:
+            try:
+                data_product = db.get_one(DataProductModel, id)
+            except NoResultFound:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND, detail="product not founds"
+                )
+        elif data_product_name:
+            data_product = db.query(DataProductModel).get_one(
```

Comment:
I thought we decided to use a new approach to query data (SQLAlchemy 2.0), isn't it?

Reply:
Forgot about it. To me it also feels less intuitive, but it probably just takes some getting used to